theano/sandbox/cuda/elemwise.py

```diff
@@ -132,9 +132,9 @@ def c_src_kernel(self, node, nodename, nd):
         #TODO: What if the scalar_op needs support_code??
         task_code = self.scalar_op.c_code(
             Apply(self.scalar_op,
-                  [scalar.Scalar(dtype=input.type.dtype)()
+                  [scalar.Scalar(dtype=input.type.dtype).make_variable()
                    for input in node.inputs],
-                  [scalar.Scalar(dtype=output.type.dtype)()
+                  [scalar.Scalar(dtype=output.type.dtype).make_variable()
```

Comment:
For this, we can just set the stack size to 0. It is equivalent. I did that
at a few place. But elemwise c code is called from many place just to know
if we can generate c code. I do not expect this to be needed for other ops.

On Tue, Feb 3, 2015 at 10:46 AM, abergeron <notifications@github.com> wrote:

> In theano/sandbox/cuda/elemwise.py
> <https://github.com/Theano/Theano/pull/2460#discussion_r24012428>:
>
> >                     for input in node.inputs],
> > -                  [scalar.Scalar(dtype=output.type.dtype)()
> > +                  [scalar.Scalar(dtype=output.type.dtype).make_variable()
>
> Then we should have a global flag to disable extract_stack() and not
> change this kind of thing everywhere.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/2460/files#r24012428>.
>